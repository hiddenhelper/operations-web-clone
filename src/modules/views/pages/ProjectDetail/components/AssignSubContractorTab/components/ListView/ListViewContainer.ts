import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import { GeneralModel, ProjectModel } from '../../../../../../../models';
import { GENERAL } from '../../../../../../../../constants';
import { IRootState } from '../../../../../../../state-mgmt/rootState';
import { clientState } from '../../../../../../../state-mgmt/client';

import ListView from './ListView';

export const mapStateToProps = (state: IRootState) => ({
  clientMap: state.client.clientProjectMap,
  clientCount: state.client.count,
  userRole: state.auth.role,
  mwbeList: state.client.mwbeList,
  clientLoading: state.general.loadingMap[GENERAL.LOADING_KEY.FETCH_PROJECT_CLIENT_LIST],
  assignClientLoading: state.general.loadingMap[GENERAL.LOADING_KEY.ASSIGN_CLIENT_PROJECT],
  projectClientSummaryLoading: state.general.loadingMap[GENERAL.LOADING_KEY.FETCH_PROJECT_CLIENT_SUMMARY],
  taxConditionLoading: state.general.loadingMap[GENERAL.LOADING_KEY.UPDATE_PROJECT_CLIENT_TAX_CONDITION],
});

export const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchProjectClientList: (id: string, query: GeneralModel.IQueryParams) => dispatch(clientState.actions.fetchProjectClientListStart(id, query)),
  fetchProjectClientSummary: (projectId: string, clientId: string) => dispatch(clientState.actions.fetchProjectClientSummaryStart(projectId, clientId)),
  updateTaxCondition: (projectId: string, taxCondition: ProjectModel.IClientTaxCondition) =>
    dispatch(clientState.actions.updateProjectClientTaxConditionStart(projectId, taxCondition)),
  fetchMwbe: () => dispatch(clientState.actions.fetchMWbeListStart()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ListView);
