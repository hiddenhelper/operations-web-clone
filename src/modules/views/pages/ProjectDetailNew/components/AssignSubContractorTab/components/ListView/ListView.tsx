import React, { memo, useMemo, useEffect, useRef, useCallback, useState } from 'react';
import { Table, TableBody, TableCell, TableRow, TableHead, withStyles } from '@material-ui/core';

import ClientDrawer from 'modules/views/shared/ClientDrawer';
import EditTaxesModal from 'modules/views/shared/ClientDrawer/EditTaxesModal';
import Modal from 'modules/views/shared/Modal';
import Pagination from 'modules/views/shared/Pagination';
import StatusChip from 'modules/views/shared/StatusChip';
import TableCellLink from 'modules/views/shared/TableCellLink';
import PermissionGuard from 'modules/views/shared/PermissionGuard';

import { GeneralModel, ClientModel, ProjectModel, UserModel } from 'modules/models';
import { stateMap } from 'constants/index';
import { getTradesString } from 'utils/tradeUtils';
import { useStyles as statusChipStyles } from '../../../../../../shared/StatusChip/styles';
import { useStyles as modalStyles } from '../../../../../../shared/Modal/style';
import { useStyles, tableRowStyles } from '../../../../styles';
import { hasValidPermissions } from 'modules/models/user';

const StyledTableRow = withStyles(tableRowStyles)(TableRow);

export interface IListViewProps {
  projectId: string;
  queryParams: GeneralModel.IQueryParams;
  clientMap: GeneralModel.IEntityMap<GeneralModel.IEntityMap<ClientModel.IClientProject>>;
  clientCount: number;
  mwbeList: GeneralModel.INamedEntity[];
  clientLoading: GeneralModel.ILoadingStatus;
  assignClientLoading: GeneralModel.ILoadingStatus;
  projectClientSummaryLoading: GeneralModel.ILoadingStatus;
  taxConditionLoading: GeneralModel.ILoadingStatus;
  drawer: { open: boolean; id: string };
  currentUserPermissions: UserModel.IPermission[];
  setDrawer: ({ open: boolean, id: string }) => void;
  closeModal: () => void;
  onPageChange: (page: number) => void;
  fetchProjectClientList: (id: string, query: GeneralModel.IQueryParams) => void;
  fetchProjectClientSummary: (projectId: string, companyId: string) => void;
  updateTaxCondition: (projectId: string, taxCondition: ProjectModel.IClientTaxCondition) => void;
  fetchMwbe: () => void;
  isFcaUser: boolean;
  isAdmin: boolean;
}

const ListView = ({
  projectId,
  queryParams,
  clientMap,
  clientCount,
  mwbeList,
  clientLoading,
  assignClientLoading,
  projectClientSummaryLoading,
  taxConditionLoading,
  drawer,
  currentUserPermissions,
  isFcaUser,
  isAdmin,
  setDrawer,
  closeModal,
  onPageChange,
  fetchProjectClientList,
  fetchProjectClientSummary,
  updateTaxCondition,
  fetchMwbe,
}: IListViewProps) => {
  const classes = useStyles();
  const statusChipClasses = statusChipStyles();
  const modalClasses = modalStyles();
  const clientListRef = useRef();
  const [showEditTaxModal, setShowEditTaxModal] = useState(false);

  const countClients = useMemo(() => Math.ceil(clientCount / queryParams.limit), [clientCount, queryParams.limit]);
  const clientList = useMemo(() => (projectId && Object.keys(clientMap).length && clientMap[projectId] ? Object.values(clientMap[projectId]) : []), [
    clientMap,
    projectId,
  ]);

  const currentClient = useMemo(
    /* istanbul ignore next */ () =>
      projectId && Object.keys(clientMap).length && clientMap[projectId] && clientMap[projectId][drawer?.id]
        ? clientMap[projectId][drawer?.id]
        : { ...ClientModel.getFallbackClient(), isTaxExempt: false },
    [projectId, clientMap, drawer]
  );

  const clientProjectStatusMap = useMemo(
    () => ({
      [ProjectModel.CompanyProjectStatus.PENDING]: { label: 'Pending', class: statusChipClasses.pending },
      [ProjectModel.CompanyProjectStatus.ACCEPTED]: { label: 'Accepted', class: statusChipClasses.active },
      [ProjectModel.CompanyProjectStatus.REJECTED]: { label: 'Rejected', class: statusChipClasses.expired },
    }),
    [statusChipClasses]
  );

  const onOpenDrawer = useCallback(
    (clientId: string) => {
      fetchProjectClientSummary(projectId, clientId);
      setDrawer({ open: true, id: clientId });
    },
    [setDrawer, fetchProjectClientSummary, projectId]
  );
  const onCloseDrawer = useCallback(/* istanbul ignore next */ () => setDrawer({ open: false, id: null }), [setDrawer]);

  const closeTaxesModal = useCallback(/* istanbul ignore next */ () => setShowEditTaxModal(false), [setShowEditTaxModal]);

  const handleEditTaxButtonClick = useCallback(() => {
    setShowEditTaxModal(true);
  }, [setShowEditTaxModal]);

  const handleEditTaxConfirm = useCallback(
    condition => {
      updateTaxCondition(projectId, { companyId: currentClient.id, isTaxExempt: condition });
    },
    [updateTaxCondition, projectId, currentClient.id]
  );

  useEffect(() => {
    /* istanbul ignore else */
    if (!mwbeList.length) fetchMwbe();
  }, [mwbeList, fetchMwbe]);

  useEffect(() => {
    /* istanbul ignore else */
    if (projectId) fetchProjectClientList(projectId, queryParams);
  }, [projectId, queryParams, fetchProjectClientList]);

  useEffect(() => {
    /* istanbul ignore else */
    if (assignClientLoading && !assignClientLoading.isLoading) {
      closeModal();
      fetchProjectClientList(projectId, queryParams);
    }
  }, [assignClientLoading, projectId, queryParams, closeModal, fetchProjectClientList]);

  useEffect(() => {
    if (taxConditionLoading && !taxConditionLoading.isLoading) {
      setShowEditTaxModal(false);
    }
  }, [taxConditionLoading]);

  useEffect(() => {
    return function unMount() {
      onCloseDrawer();
    };
  }, [onCloseDrawer]);

  if (clientLoading && clientLoading.isLoading) {
    return <>Loading...</>;
  }

  return (
    <>
      <Table aria-label="client-list">
        <TableHead>
          <TableRow>
            <TableCell>Name</TableCell>
            <TableCell>Role in Project</TableCell>
            <TableCell>Location</TableCell>
            <TableCell>Trades</TableCell>
            <TableCell>Status in Project</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {clientList.map(client => (
            <StyledTableRow
              data-testid="client-list-row"
              key={client.id}
              onClick={() => (hasValidPermissions(UserModel.ClientsPermission.VIEWACCESS, currentUserPermissions) ? onOpenDrawer(client.id) : '')}
              className={classes.pointer}
            >
              <TableCell data-testid="project-client-row">
                <PermissionGuard shouldbeFCAUser={true} permissionsExpression={UserModel.ClientsPermission.VIEWACCESS} fallback={<>{client.name}</>}>
                  <TableCellLink href={`/clients/detail/${client.id}`} text={client.name} title="View Client details" />
                </PermissionGuard>
              </TableCell>
              <TableCell>{ProjectModel.roleMap[client.role]}</TableCell>
              <TableCell>
                {client?.jobSiteAddress?.city ? (
                  <>
                    {client.jobSiteAddress.city}, {stateMap[client.jobSiteAddress.stateCode]}
                  </>
                ) : (
                  '-'
                )}
              </TableCell>
              <TableCell>{getTradesString(client.trades, client.otherTrade)}</TableCell>
              <TableCell>
                <StatusChip
                  styleClasses={`${clientProjectStatusMap[client.companyProjectStatus].class} `}
                  label={clientProjectStatusMap[client.companyProjectStatus].label}
                />
              </TableCell>
            </StyledTableRow>
          ))}
        </TableBody>
      </Table>
      <Pagination page={queryParams.page} count={countClients} onChange={onPageChange} />
      <PermissionGuard permissionsExpression={UserModel.ClientsPermission.VIEWACCESS}>
        <ClientDrawer
          isOpen={drawer?.open}
          isLoading={projectClientSummaryLoading?.isLoading}
          showTaxExempt={true}
          showPrimaryButton={isFcaUser}
          showSecondaryButton={false}
          buttonText="Edit Taxes"
          buttonTestId="drawerEditTaxButton"
          client={currentClient}
          clientListElement={clientListRef.current}
          mwbeList={mwbeList}
          onClose={onCloseDrawer}
          handleButtonClick={handleEditTaxButtonClick}
        />
      </PermissionGuard>
      <PermissionGuard permissionsExpression={UserModel.ClientProjectsPermission.MANAGE}>
        <Modal
          show={showEditTaxModal}
          styleClass={`${modalClasses.dialogContainer} ${modalClasses.deleteModal} ${classes.taxPopUpContainer}`}
          render={() => (
            <EditTaxesModal
              taxCondition={currentClient.isTaxExempt}
              loading={taxConditionLoading?.isLoading}
              onCancel={closeTaxesModal}
              onConfirm={handleEditTaxConfirm}
            />
          )}
        />
      </PermissionGuard>
    </>
  );
};

export default memo(ListView);
