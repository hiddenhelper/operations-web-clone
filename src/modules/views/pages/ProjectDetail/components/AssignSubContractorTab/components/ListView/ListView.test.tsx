import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';
import { MemoryRouter } from 'react-router';

import { getProject_1, getClientProject_1, getClientProject_2, getDefaultLoading, getMwbeType_1 } from '../../../../../../../../test/entities';
import { getInitialState } from '../../../../../../../../test/rootState';
import ListView, { IListViewProps } from './ListView';
import { UserModel } from '../../../../../../../models';

jest.useFakeTimers();

describe('ListView', () => {
  let props: IListViewProps;

  beforeEach(() => {
    props = {
      projectId: getProject_1().id,
      queryParams: { page: 1, limit: 15 },
      mwbeList: getMwbeType_1(),
      clientMap: {
        [getProject_1().id]: {
          [getClientProject_1().id]: getClientProject_1(),
          [getClientProject_2().id]: getClientProject_2(),
        },
      },
      userRole: UserModel.Role.FCA_ADMIN,
      drawer: { open: false, id: null },
      clientCount: 16,
      assignClientLoading: getDefaultLoading(),
      clientLoading: getDefaultLoading(),
      projectClientSummaryLoading: getDefaultLoading(),
      taxConditionLoading: getDefaultLoading(),
      setDrawer: jest.fn(),
      closeModal: jest.fn(),
      onPageChange: jest.fn(),
      fetchProjectClientList: jest.fn(),
      fetchProjectClientSummary: jest.fn(),
      updateTaxCondition: jest.fn(),
      fetchMwbe: jest.fn(),
    };
  });

  it('should render', () => {
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <ListView {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render loading', () => {
    props.clientLoading = { isLoading: true, hasError: false, error: null };
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <ListView {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should change clients page', () => {
    const { getByText } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <ListView {...props} />
        </MemoryRouter>
      </Provider>
    );

    const pageItem = getByText('2');

    act(() => {
      fireEvent.click(pageItem);
    });

    act(() => {
      fireEvent.click(pageItem);
    });

    expect(props.onPageChange).toHaveBeenCalledWith(2);
  });

  it('should update tax condition', () => {
    props.clientMap = {
      [getProject_1().id]: {
        [getClientProject_1().id]: { ...getClientProject_1(), isTaxExempt: true },
        [getClientProject_2().id]: { ...getClientProject_2(), isTaxExempt: true },
      },
    };
    const wrapper = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <ListView {...props} />
        </MemoryRouter>
      </Provider>
    );

    act(() => {
      fireEvent.click(wrapper.getAllByTestId('project-client-row')[0]);
    });

    expect(props.fetchProjectClientSummary).toHaveBeenCalled();

    act(() => {
      fireEvent.click(wrapper.getByTestId('drawerEditTaxButton'));
    });

    act(() => {
      fireEvent.mouseDown(wrapper.getByText('Not Exempt'));
    });

    act(() => {
      jest.runOnlyPendingTimers();
    });

    act(() => {
      fireEvent.click(wrapper.getAllByRole('option')[1]);
    });

    act(() => {
      fireEvent.click(wrapper.getByText('Save'));
    });

    expect(props.updateTaxCondition).toHaveBeenCalled();
  });

  it('should render client info', () => {
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <ListView {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
