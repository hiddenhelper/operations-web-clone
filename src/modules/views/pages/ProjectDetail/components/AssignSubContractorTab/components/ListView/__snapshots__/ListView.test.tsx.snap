// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView should render 1`] = `
<div>
  <table
    aria-label="client-list"
    class="MuiTable-root"
  >
    <thead
      class="MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Role in Project
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Location
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Trades
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Status in Project
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-114 makeStyles-pointer-16"
        data-testid="client-list-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
          data-testid="project-client-row"
        >
          Robert C. Martin
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          Developer
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          New York
          , 
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          trade 1, trade 2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          <div
            class="MuiChip-root makeStyles-statusChip-77 makeStyles-pending-84  MuiChip-colorPrimary"
          >
            <span
              class="MuiChip-label"
            >
              Pending
            </span>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-114 makeStyles-pointer-16"
        data-testid="client-list-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
          data-testid="project-client-row"
        >
          Martin Fowler
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          General Contractor
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          -
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          trade 1, trade 2, trade 3, some trade
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          <div
            class="MuiChip-root makeStyles-statusChip-77 makeStyles-active-83  MuiChip-colorPrimary"
          >
            <span
              class="MuiChip-label"
            >
              Accepted
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root  makeStyles-pagination-115"
    data-testid="pagination"
  >
    <ul
      class="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-116 makeStyles-close-127 closed"
    data-testid="client-drawer-detail"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-117 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
    >
      <div
        class="makeStyles-drawerButtonWrapper-119"
      >
        <div
          class="makeStyles-drawerButton-120"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="drawer-close-btn"
            style="outline: none;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg>
                close.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-drawerHeader-118"
        >
          <p
            class="MuiTypography-root makeStyles-drawerTitle-121 MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignLeft"
          >
            Client Information
          </p>
        </div>
      </div>
      <div
        class="makeStyles-listDetail-146"
      >
        <div
          class="makeStyles-company-147"
        >
          <div
            class="makeStyles-drawerText-123"
          >
            <p
              class="MuiTypography-root makeStyles-listAccent-148 makeStyles-entityTitle-145 MuiTypography-body1"
            />
            <div
              class="makeStyles-drawerMainText-124"
            >
              <span>
                -
              </span>
              <span>
                -
              </span>
              <span />
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-122"
          >
            <svg>
              location.svg
            </svg>
            <div
              class="makeStyles-drawerText-123 makeStyles-drawerTextSeparation-125"
            >
              <p
                class="MuiTypography-root makeStyles-cardFont-185  MuiTypography-body1"
              >
                
                <span
                  class=" "
                >
                  -
                </span>
              </p>
              <p
                class="MuiTypography-root makeStyles-cardFont-185  MuiTypography-body1"
              >
                
                
                
                
              </p>
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-122"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
            <div
              class="makeStyles-drawerText-123 makeStyles-drawerTextSeparation-125"
            >
              <p
                class="MuiTypography-root makeStyles-cardFont-185  MuiTypography-body1"
              >
                
                <span
                  class=" "
                >
                  -
                </span>
              </p>
              <p
                class="MuiTypography-root makeStyles-cardFont-185  MuiTypography-body1"
              >
                
                
                
                
              </p>
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-122"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <div
              class="makeStyles-drawerText-123 makeStyles-drawerTextSeparation-125"
            >
              <span
                class="makeStyles-listAccent-148"
              >
                -
              </span>
              <span>
                -
              </span>
              <span>
                -
              </span>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root makeStyles-drawerDivider-126"
        />
        <div
          class="makeStyles-ctaWrapper-149"
        >
          <span
            class=""
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-165 makeStyles-borderPrimaryButton-157"
              data-testid="drawerEditTaxButton"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Edit Taxes
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ListView should render client info 1`] = `
<div>
  <table
    aria-label="client-list"
    class="MuiTable-root"
  >
    <thead
      class="MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Role in Project
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Location
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Trades
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Status in Project
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-834 makeStyles-pointer-736"
        data-testid="client-list-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
          data-testid="project-client-row"
        >
          Robert C. Martin
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          Developer
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          New York
          , 
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          trade 1, trade 2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          <div
            class="MuiChip-root makeStyles-statusChip-797 makeStyles-pending-804  MuiChip-colorPrimary"
          >
            <span
              class="MuiChip-label"
            >
              Pending
            </span>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-834 makeStyles-pointer-736"
        data-testid="client-list-row"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
          data-testid="project-client-row"
        >
          Martin Fowler
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          General Contractor
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          -
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          trade 1, trade 2, trade 3, some trade
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          <div
            class="MuiChip-root makeStyles-statusChip-797 makeStyles-active-803  MuiChip-colorPrimary"
          >
            <span
              class="MuiChip-label"
            >
              Accepted
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root  makeStyles-pagination-835"
    data-testid="pagination"
  >
    <ul
      class="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page Mui-selected"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-836 makeStyles-close-847 closed"
    data-testid="client-drawer-detail"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-837 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
    >
      <div
        class="makeStyles-drawerButtonWrapper-839"
      >
        <div
          class="makeStyles-drawerButton-840"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="drawer-close-btn"
            style="outline: none;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg>
                close.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-drawerHeader-838"
        >
          <p
            class="MuiTypography-root makeStyles-drawerTitle-841 MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignLeft"
          >
            Client Information
          </p>
        </div>
      </div>
      <div
        class="makeStyles-listDetail-866"
      >
        <div
          class="makeStyles-company-867"
        >
          <div
            class="makeStyles-drawerText-843"
          >
            <p
              class="MuiTypography-root makeStyles-listAccent-868 makeStyles-entityTitle-865 MuiTypography-body1"
            />
            <div
              class="makeStyles-drawerMainText-844"
            >
              <span>
                -
              </span>
              <span>
                -
              </span>
              <span />
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-842"
          >
            <svg>
              location.svg
            </svg>
            <div
              class="makeStyles-drawerText-843 makeStyles-drawerTextSeparation-845"
            >
              <p
                class="MuiTypography-root makeStyles-cardFont-905  MuiTypography-body1"
              >
                
                <span
                  class=" "
                >
                  -
                </span>
              </p>
              <p
                class="MuiTypography-root makeStyles-cardFont-905  MuiTypography-body1"
              >
                
                
                
                
              </p>
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-842"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
            <div
              class="makeStyles-drawerText-843 makeStyles-drawerTextSeparation-845"
            >
              <p
                class="MuiTypography-root makeStyles-cardFont-905  MuiTypography-body1"
              >
                
                <span
                  class=" "
                >
                  -
                </span>
              </p>
              <p
                class="MuiTypography-root makeStyles-cardFont-905  MuiTypography-body1"
              >
                
                
                
                
              </p>
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-842"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <div
              class="makeStyles-drawerText-843 makeStyles-drawerTextSeparation-845"
            >
              <span
                class="makeStyles-listAccent-868"
              >
                -
              </span>
              <span>
                -
              </span>
              <span>
                -
              </span>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root makeStyles-drawerDivider-846"
        />
        <div
          class="makeStyles-ctaWrapper-869"
        >
          <span
            class=""
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-885 makeStyles-borderPrimaryButton-877"
              data-testid="drawerEditTaxButton"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Edit Taxes
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ListView should render loading 1`] = `
<div>
  Loading...
</div>
`;
